JS - reset correctly radio buttons

calixteman opened the pull request.

calixteman committed:
JS - reset correctly radio buttons

Snuffleupagus commented on src/display/annotation_layer.js:
```diff
@@ -1016,19 +1015,16 @@ class RadioButtonWidgetAnnotationElement extends WidgetAnnotationElement {
-                storage.setValue(radioId, { value: false });
-              }
+              radio.checked = radioId === id && checked;
+              storage.setValue(id, { value: radio.checked });
```
Unless I'm completely misunderstanding this line, won't it simply keep overwriting the *same* AnnotationStorage entry over and over for each loop iteration?

Is that actually intended, or should it still say `radioId` here?

calixteman replied:
pfff... thx

timvandermeij commented:
/botio test

timvandermeij commented:
It looks like there is a regression in the `annotation-button-widget-forms-page1` test. I can only guess that this has something to do with the `setAttribute` change, although I don't really understand why, since the other changes are related to scripting which that file doesn't use...

timvandermeij commented:
It looks like there is a regression in some checkbox reference tests. I can only guess that this has something to do with the setAttribute change, although I don't really understand why, since the other changes are related to scripting which that file doesn't use...

Snuffleupagus commented:
>  I can only guess that this has something to do with the `setAttribute` change,

If that's case, perhaps this comment applies here as well (meaning that we need to somehow fix/update the code in `test/driver.js`): https://github.com/mozilla/pdf.js/blob/709c57a66b176d17db4b2e034d16daddd3860f33/src/display/annotation_layer.js#L636-L638

calixteman commented:
> It looks like there is a regression in some checkbox reference tests. I can only guess that this has something to do with the `setAttribute` change, although I don't really understand why, since the other changes are related to scripting which that file doesn't use...

I replaced `radio.setAttribute("checked"...` in the listener because else it doesn't work and so to be consistent I changed the other one too.

calixteman commented:
/botio test

Snuffleupagus commented on src/display/annotation_layer.js:
```diff
@@ -915,7 +915,7 @@ class CheckboxWidgetAnnotationElement extends WidgetAnnotationElement {
     element.name = this.data.fieldName;
     if (value) {
-      element.setAttribute("checked", true);
+      element.setAttributes("checked", true);
```
There's a extra "s" added at the end of the method, which probably explains why not all tests run with the test logs containing:
```
TEST-UNEXPECTED-FAIL | test failed firefox has not responded in 120s
TEST-UNEXPECTED-FAIL | test failed chrome has not responded in 120s
```

Snuffleupagus commented on src/display/annotation_layer.js:
```diff
@@ -1000,9 +1000,8 @@ class RadioButtonWidgetAnnotationElement extends WidgetAnnotationElement {
     element.name = data.fieldName;
     if (value) {
-      element.setAttribute("checked", true);
+      element.setAttributes("checked", true);
```
As above.

timvandermeij commented:
/botio test

timvandermeij commented:
/botio integrationtest

timvandermeij commented:
Thanks!
